clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesNamedAccountCredentials.groovy

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.kubernetes.security
+
+import com.netflix.spinnaker.clouddriver.kubernetes.api.KubernetesApiAdaptor
+import com.netflix.spinnaker.clouddriver.kubernetes.config.LinkedDockerRegistryConfiguration
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsRepository
+import io.fabric8.kubernetes.client.Config
+
+public class KubernetesNamedAccountCredentials implements AccountCredentials<KubernetesCredentials> {
+  static final String CLOUD_PROVIDER = "kubernetes"
+  final String accountName
+  final String environment
+  final String accountType
+  final String context
+  final String cluster
+  final String user
+  final String userAgent
+  final String kubeconfigFile
+  List<String> namespaces
+  KubernetesCredentials credentials
+  final List<String> requiredGroupMembership
+  final List<LinkedDockerRegistryConfiguration> dockerRegistries
+  private final AccountCredentialsRepository accountCredentialsRepository
+
+  public KubernetesNamedAccountCredentials(AccountCredentialsRepository accountCredentialsRepository,
+                                           String userAgent,
+                                           String accountName,
```

Comment:
Personal preference, really, but I'd expect Kubernetes**Named**AccountCreds to have _name_ as its first arg.

Secondly, I stripped _account_name to be just name, since the "account" part is implied by the class.